### PR TITLE
Rbac and namespaces

### PR DESCRIPTION
**Pull request by cavemanpi**

### CODE REVIEW

File: lib/Net/Kubernetes.pm [hunk 1]
```diff
@@ -172,6 +177,57 @@ sub get_namespace {
...
+    my ($self, $namespace) = @_;
+
+    if (!defined $namespace || !length $namespace) {
+        Throwable::Error->throw(message => '$namespace cannot be null');
+    }
+
+    my $namespace_path = $self->get_resource_path('namespace');
+    my $res = $self->ua->request($self->create_request(
+        POST => $namespace_path, 
+        undef, $self->json->encode({
```
Comment (jimbobhickville) [inline lib/Net/Kubernetes.pm]: Just a style nit, but maybe a newline between `undef` and the json body would make this not read quite so awkwardly.

File: lib/Net/Kubernetes/Role/APIAccess.pm [hunk 2]
```diff
@@ -157,17 +158,23 @@ sub _build_server_version {
 
 sub _build_lwp_agent {
     my $self = shift;
-    my $ua = LWP::UserAgent->new(agent => 'net-kubernetes-perl/0.20');
+
+    my $ua;
+    my %ua_args = (
+	agent    => 'net-kubernetes-perl/1.06',
```
Comment (jimbobhickville) [inline lib/Net/Kubernetes/Role/APIAccess.pm]: Whitespace doesn't line up here.

**Review by jimbobhickville — APPROVED** [review]
lgtm to me. A couple really minor style nits. Is `list_service_accounts` no longer a thing?

**Comment by cavemanpi** [comment]
Service accounts are not global resources but were previously written as though they are. It's list method has been moved to the resource lister role. The resource lister role is consumed by the namespace class. Net::Kubernetes delegates list_service_accounts to the default namespace.

**Pull request merged**